app/workers/activitypub/fetch_all_replies_worker.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+# Fetch all replies to a status, querying recursively through
+# ActivityPub replies collections, fetching any statuses that
+# we either don't already have or we haven't checked for new replies
+# in the Status::FETCH_REPLIES_DEBOUNCE interval
+class ActivityPub::FetchAllRepliesWorker
+  include Sidekiq::Worker
+  include ExponentialBackoff
+  include JsonLdHelper
+
+  sidekiq_options queue: 'pull', retry: 3
+
+  # Global max replies to fetch per request (all replies, recursively)
+  MAX_REPLIES = (ENV['FETCH_REPLIES_MAX_GLOBAL'] || 1000).to_i
+  MAX_PAGES = (ENV['FETCH_REPLIES_MAX_PAGES'] || 500).to_i
+
+  def perform(parent_status_id, options = {})
+    @parent_status = Status.find(parent_status_id)
+    Rails.logger.debug { "FetchAllRepliesWorker - #{@parent_status.uri}: Fetching all replies for status: #{@parent_status}" }
+
+    uris_to_fetch, n_pages = get_replies(@parent_status.uri, MAX_PAGES, options)
+    return if uris_to_fetch.nil?
+
+    @parent_status.touch(:fetched_replies_at)
```

Comment:
```suggestion
    return unless @parent_status.should_fetch_replies?
    @parent_status.touch(:fetched_replies_at)
    Rails.logger.debug { "FetchAllRepliesWorker - #{@parent_status.uri}: Fetching all replies for status: #{@parent_status}" }

    uris_to_fetch, n_pages = get_replies(@parent_status.uri, MAX_PAGES, options)
    return if uris_to_fetch.nil?

```

imperfectly prevent a race condition where multiple api calls would've bypassed cooldown, full fix requires redis lock

Reply:
This is currently handled by the outer context scope, I am not sure how expensive the "`should_fetch_replies?`" lookup is but this would mean we do it twice. If we did this then it seems like the parent call should just call the worker always on a call to `context`, since the worker handles quitting early if replies shouldn't be fetched. The reason we might not want to do that is that the outer context method eventually should know whether we are actually doing the fetch so that it can signal more replies will come in future calls. 

So for now i'll just accept this and we can work out that later - i dont think the double call to the `should_fetch_replies?` should be that expensive.